android/src/main/java/io/ably/lib/push/ActivationStateMachine.java

```diff
@@ -678,21 +717,12 @@ private boolean persist() {
         for (int i = 0; i < length; i++) {
             try {
                 String className = activationContext.getPreferences().getString(String.format("%s[%d]", ActivationStateMachine.PersistKeys.PENDING_EVENTS_PREFIX, i), "");
-                Class<ActivationStateMachine.Event> eventClass = (Class<ActivationStateMachine.Event>) Class.forName(className);
-                ActivationStateMachine.Event event;
-                try {
-                    event = eventClass.newInstance();
-                } catch(InstantiationException e) {
-                    // We aren't properly persisting events with a non-nullary constructor. Those events
-                    // are supposed to be handled by states that aren't persisted (until
-                    // https://github.com/ably/ably-java/issues/546 is fixed), so it should be safe to
-                    // just drop them.
-                    Log.d(TAG, String.format("discarding improperly persisted event: %s", className));
-                    continue;
-                }
+                ActivationStateMachine.Event event = Event.constructEventByName(className);
                 deque.add(event);
-            } catch(Exception e) {
-                throw new RuntimeException(e);
+            } catch (ClassNotFoundException e) {
+                Log.e(TAG, e.getLocalizedMessage());
+            } catch (InstantiationException e) {
+                continue;
```

Comment:
If this needs to be checked then is there any reason we're not logging it out also?